MatrixKit/Models/Room/MXKRoomBubbleCellDataWithAppendingMode.m

```diff
@@ -141,33 +145,37 @@ - (void)prepareBubbleComponentsPosition
     // Set position of the first component
     [super prepareBubbleComponentsPosition];
     
-    // Check whether the position of other components need to be refreshed
-    if (self.attachment || !shouldUpdateComponentsPosition || bubbleComponents.count < 2)
+    @synchronized(bubbleComponents)
     {
-        return;
-    }
-    
-    // Compute height of the first text component
-    MXKRoomBubbleComponent *component = [bubbleComponents firstObject];
-    CGFloat componentHeight = [self rawTextHeight:component.attributedTextMessage];
-    
-    // Set position for each other component
-    CGFloat positionY = component.position.y;
-    CGFloat cumulatedHeight = 0;
-    NSMutableAttributedString *attributedString = [[NSMutableAttributedString alloc] initWithAttributedString:component.attributedTextMessage];
-    for (NSUInteger index = 1; index < bubbleComponents.count; index++)
-    {
-        cumulatedHeight += componentHeight;
-        positionY += componentHeight;
+        // Check whether the position of other components need to be refreshed
+        if (self.attachment || !shouldUpdateComponentsPosition || bubbleComponents.count < 2)
+        {
+            return;
```

Comment:
In order to avoid this return, the test  should be:
if (!self.attachment && shouldUpdateComponentsPosition && bubbleComponents.count > 1)